sklearn/neighbors/graph.py

```diff
@@ -77,7 +79,13 @@ def kneighbors_graph(X, n_neighbors, mode='connectivity', metric='minkowski',
             ).fit(X)
     else:
         _check_params(X, metric, p, metric_params)
-    return X.kneighbors_graph(X._fit_X, n_neighbors, mode=mode)
+    if mode == "connectivity":
+        warnings.warn("The behavior of marking each sample itself as the "
```

Comment:
The only way I could think of is adding an additional parameter, but then we would have to deprecate the parameter again. That seems overly complicated, and means the user needs to change their code twice, once adding the ``i_want_the_fix=True`` and then removing it again.